config/job.js

```diff
@@ -168,6 +178,7 @@ const SCHEMA_JOB = Joi.object()
         requires: SCHEMA_REQUIRES,
         secrets: SCHEMA_SECRETS,
         settings: SCHEMA_SETTINGS,
+        stage: SCHEMA_STAGE,
```

Comment:
I believe the `jobs` attribute inside `stage` definition in `screwdriver.yaml` is the source of truth to associate a `job` to a `stage`. Having `stage` inside `job` configuration would be redundant and needs additional validation to ensure it point to the correct `stage`.
We should not allow it to be configurable under `job`. 

```suggestion
```